src/acl/Asn.cc

```diff
@@ -246,6 +246,7 @@ asnCacheStart(int as)
     asState->request = HttpRequest::FromUrl(asres, mx);
     assert(asState->request != NULL);
 
+    // XXX: Missing a hittingRequiresCollapsing() && startCollapsingOn() check.
```

Comment:
> what do you mean exactly by this comment?

This comment is about a general collapsing 'rule': since each of storeGetPublic* methods can return a collapsed entry we should check the result against our collapse-limiting directives, like 'collapsed_forwarding" and "collapsed_forwarding_access". For example, see one such use case clientReplyContext::processExpired().
However, I think this comment is useless here because ASN code does not initiate collapsing at all (i.e., no Controller::allowCollapsing() for this entry).

> because the ASN cache generating the lookup should collapse earlier on the ASM number lookup level itself

I have not found any code related to collapsing at ASN-numbers level. But, for another reason (above), we should remove this comment, I think.

Reply:
> this comment is useless here because ASN code does not initiate collapsing at all

Can some other code initiate collapsing for the same URL/entry? For example, can I request the same whois(?) URL by sending an HTTP request with that URL to Squid today? If that is not allowed today, is there some fundamental reason why it cannot be supported tomorrow?

If requesting these URLs by other channels is impossible, I would still leave a comment here to increase our chances of quickly finding bugs when the ASN code is modified to allow collapsing. It just would not be an XXX comment anymore. I can suggest specific wording if necessary.

Reply:
> For example, can I request the same whois(?) URL by sending an HTTP request with that URL to Squid today?

I tested, sending such HTTP request with "whois://host:port/path" URL, where 'path' formatted like in asnCacheStart(). In my test, Squid converted it to whois() request, sending the 'path' to the server. Collapsing, if enabled, also works for such requests. So, considering this, we should not remove this XXX.

Reply:
Agreed. The XXX correctly identifies a bug (and even hints at the solution).